### PR TITLE
fix(ZNTA-2721): batch requests to server

### PR DESCRIPTION
**Pull request by djansen-redhat**
JIRA issue URL: https://zanata.atlassian.net/browse/ZNTA-2721

Pushing 30k strings through MT proved to be excessive force. Batch to 1000.

## Checklist

- JIRA link
- Check target branch
- Make sure all commit statuses are green or otherwise documented reasons to ignore
- QA needs to evaluate against the JIRA ticket
- Changed files and commits make sense for this PR

See [Zanata Development Guidelines](https://github.com/zanata/zanata-platform/wiki/Development-Guidelines) more for information.

----
*This template can be updated in .github/PULL_REQUEST_TEMPLATE.md*

### CODE REVIEW

File: server/services/src/main/java/org/zanata/service/impl/MachineTranslationServiceImpl.java [hunk 1]
```diff
@@ -86,6 +86,7 @@
 public class MachineTranslationServiceImpl implements
         MachineTranslationService {
     private static final int BATCH_SIZE = 100;
```
Comment (seanf) [inline server/services/src/main/java/org/zanata/service/impl/MachineTranslationServiceImpl.java]: I guess this is SAVE_BATCH_SIZE now?

Comment (djansen-redhat) [reply server/services/src/main/java/org/zanata/service/impl/MachineTranslationServiceImpl.java]: Done

File: server/services/src/main/java/org/zanata/service/impl/MachineTranslationServiceImpl.java [hunk 2]
```diff
@@ -271,17 +272,29 @@ private MTDocument getTranslationFromMT(MTDocument request,
...
+            MTDocument mtDocument = textFlowsToMTDoc
+                    .fromTextFlows(projectSlug, versionSlug,
+                            doc.getDocId(), doc.getSourceLocaleId(),
+                            next,
+                            TextFlowsToMTDoc::extractPluralIfPresent,
+                            backendId);
+            MTDocument result = getTranslationFromMT(mtDocument,
+                    targetLocale.getLocaleId());
+            saveTranslationsInBatches(next, result, targetLocale, saveState);
+            backendIdConfirmation = result.getBackendId();
```
Comment (seanf) [inline server/services/src/main/java/org/zanata/service/impl/MachineTranslationServiceImpl.java]: // TODO we only return the backendId from the final batch

I think the problem will go away with https://zanata.atlassian.net/browse/ZNTA-2732

**Comment by djansen-redhat** [comment]
___
*[server/services/src/main/java/org/zanata/service/impl/MachineTranslationServiceImpl.java, line 291 at r3](https://reviewable.io/reviews/zanata/zanata-platform/1004#-LIZoF1Qhcy2EdzEcxvk-r3-291:-LItQLOa4bw5azV-8gox:b-bibs94) ([raw file](https://github.com/zanata/zanata-platform/blob/7349e5e3c7352f610cc2f08901ca9698e8da17e3/server/services/src/main/java/org/zanata/service/impl/MachineTranslationServiceImpl.java#L291)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

    // TODO we only return the backendId from the final batch

I think the problem will go away with https://zanata.atlassian.net/browse/ZNTA-2732
</blockquote></details>

Good idea, I'll add the comment

**Pull request merged**